api/agent/agent.go

```diff
@@ -334,23 +341,14 @@ func (a *agent) hotLauncher(ctx context.Context, callObj *call) {
 
 		curStats := callObj.slots.getStats()
 		isNeeded := isNewContainerNeeded(&curStats)
-		logger.WithFields(logrus.Fields{
-			"currentStats": curStats,
-			"isNeeded":     isNeeded,
-		}).Debug("Hot function launcher stats")
+		logger.WithFields(logrus.Fields{"currentStats": curStats, "isNeeded": isNeeded}).Debug("Hot function launcher stats")
 		if !isNeeded {
 			continue
 		}
-
-		ctxResource, cancelResource := context.WithCancel(context.Background())
```

Comment:
so, here this creates a shorter life span ctx intentionally since for-loop would fork a go routine every time. This ensures with each for-loop iteration we cleanup any go routines.

Reply:
the only exit paths are timeout, shutdown, or get resource. i think with get resource managing its own go routines now there's no longer an issue

Reply:
so, timeout (slot busy) and impossible request (huge-mem with isOpen = false) would leak here though.

Reply:
(Currently, impossible req doesn't fork a go routine, but anyways. :-) )

Reply:
https://github.com/fnproject/fn/pull/716/commits/b644bd67c6dbe801edf1158e9d984bea503798f0 -- good catch.